test/src/main/java/org/apache/accumulo/test/functional/MemoryStarvedMajCIT.java

```diff
@@ -0,0 +1,168 @@
+      cfg.setSystemProperties(sysProps);
+    }
+  }
+
+  private static final DoubleAdder MAJC_PAUSED = new DoubleAdder();
+  private static TestStatsDSink sink;
+  private static Thread metricConsumer;
+
+  @BeforeAll
+  public static void start() throws Exception {
+    sink = new TestStatsDSink();
+    metricConsumer = new Thread(() -> {
+      while (!Thread.currentThread().isInterrupted()) {
+        List<String> statsDMetrics = sink.getLines();
+        for (String line : statsDMetrics) {
+          if (Thread.currentThread().isInterrupted()) {
+            break;
+          }
+          if (line.startsWith("accumulo")) {
+            Metric metric = TestStatsDSink.parseStatsDMetric(line);
+            if (MetricsProducer.METRICS_MAJC_PAUSED.equals(metric.getName())) {
+              Double val = Double.parseDouble(metric.getValue());
+              MAJC_PAUSED.add(val);
+            }
+          }
+        }
+      }
+    });
+    metricConsumer.start();
+
+    SharedMiniClusterBase.startMiniClusterWithConfig(new MemoryStarvedITConfiguration());
+  }
+
+  @AfterAll
+  public static void stop() throws Exception {
+    SharedMiniClusterBase.stopMiniCluster();
+    sink.close();
+    metricConsumer.interrupt();
+    metricConsumer.join();
+  }
+
+  @BeforeEach
+  public void beforeEach() throws Exception {
+    // Reset the client side counters
+    MAJC_PAUSED.reset();
+  }
+
+  @Test
+  public void testMajCPauses() throws Exception {
+    String table = getUniqueNames(1)[0];
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      TableOperations to = client.tableOperations();
+      to.create(table);
+
+      // Add a small amount of data so that the MemoryConsumingIterator
+      // returns true when trying to consume all of the memory.
+      ReadWriteIT.ingest(client, 1, 1, 1, 0, table);
+
+      AtomicReference<Throwable> error = new AtomicReference<>();
+      Thread compactionThread = new Thread(() -> {
+        try {
+          to.compact(table, new CompactionConfig().setWait(false));
+        } catch (Exception e) {
+          error.set(e);
+        }
+      });
+
+      try (Scanner scanner = client.createScanner(table)) {
+
+        MemoryStarvedScanIT.consumeServerMemory(scanner, table);
+
+        Double paused = MAJC_PAUSED.doubleValue();
+        assertEquals(0, paused);
+
+        ReadWriteIT.ingest(client, 100, 100, 100, 0, table);
+        compactionThread.start();
+
+        while (paused == 0) {
+          Thread.sleep(1000);
+          paused = MAJC_PAUSED.doubleValue();
+        }
+        assertTrue(paused > 0);
+
+        MemoryStarvedScanIT.freeServerMemory(client, table);
+        compactionThread.interrupt();
+        compactionThread.join();
+        assertNull(error.get());
+        assertTrue(client.instanceOperations().getActiveCompactions().stream()
+            .filter(ac -> ac.getPausedCount() > 0).collect(Collectors.toList()).size() > 0);
```

Comment:
```suggestion
        assertTrue(client.instanceOperations().getActiveCompactions().stream()
            .filter(ac -> ac.getPausedCount() > 0).findAny().isPresent());
```
This should be functionally equivalent.

Reply:
Addressed in f29ddfb